src/exceptions.rs

```diff
@@ -703,6 +703,11 @@ impl_windows_native_exception!(
     native_doc!("WindowsError")
 );
 
+pub(crate) struct Utf8ErrorWithBytes {
+    pub(crate) err: std::str::Utf8Error,
+    pub(crate) bytes: Vec<u8>,
+}
```

Comment:
It might make sense for this enum to live in `src/err/impls.rs` closer to where it's used if the `new_err_from_utf8` implementation changes.